pkg/templates/images/images.go

```diff
@@ -301,40 +301,67 @@ func optionalResources() map[Resource]map[string]string {
 		},
 		OpenstackCSINodeDriverRegistar: {
-			"< 1.26.0":  "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.5.1",
-			">= 1.26.0": "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.6.2",
+			"1.24.x":    "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.5.1",
+			"1.25.x":    "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.5.1",
+			"1.26.x":    "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.6.2",
+			">= 1.27.0": "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.6.2",
 		},
 		OpenstackCSILivenessProbe: {
-			"< 1.26.0":  "registry.k8s.io/sig-storage/livenessprobe:v2.7.0",
-			">= 1.26.0": "registry.k8s.io/sig-storage/livenessprobe:v2.8.0",
+			"1.24.x":    "k8s.gcr.io/sig-storage/livenessprobe:v2.6.0",
+			"1.25.x":    "registry.k8s.io/sig-storage/livenessprobe:v2.7.0",
+			"1.26.x":    "registry.k8s.io/sig-storage/livenessprobe:v2.8.0",
+			">= 1.27.0": "registry.k8s.io/sig-storage/livenessprobe:v2.9.0",
 		},
 		OpenstackCSIAttacher: {
-			"< 1.26.0":  "registry.k8s.io/sig-storage/csi-attacher:v3.4.0",
-			">= 1.26.0": "registry.k8s.io/sig-storage/csi-attacher:v4.0.0",
+			"1.24.x":    "registry.k8s.io/sig-storage/csi-attacher:v3.4.0",
+			"1.25.x":    "registry.k8s.io/sig-storage/csi-attacher:v3.4.0",
+			"1.26.x":    "registry.k8s.io/sig-storage/csi-attacher:v4.0.0",
+			">= 1.27.0": "registry.k8s.io/sig-storage/csi-attacher:v4.2.0",
 		},
 		OpenstackCSIProvisioner: {
-			"< 1.26.0":  "registry.k8s.io/sig-storage/csi-provisioner:v3.1.0",
-			">= 1.26.0": "registry.k8s.io/sig-storage/csi-provisioner:v3.4.0",
+			"1.24.x":    "registry.k8s.io/sig-storage/csi-provisioner:v3.1.0",
+			"1.25.x":    "registry.k8s.io/sig-storage/csi-provisioner:v3.1.0",
+			"1.26.x":    "registry.k8s.io/sig-storage/csi-provisioner:v3.4.0",
+			">= 1.27.0": "registry.k8s.io/sig-storage/csi-provisioner:v3.4.1",
 		},
 		OpenstackCSIResizer: {
-			"< 1.26.0":  "registry.k8s.io/sig-storage/csi-resizer:v1.4.0",
-			">= 1.26.0": "registry.k8s.io/sig-storage/csi-resizer:v1.6.0",
+			"1.24.x":    "registry.k8s.io/sig-storage/csi-resizer:v1.4.0",
+			"1.25.x":    "registry.k8s.io/sig-storage/csi-resizer:v1.4.0",
+			"1.26.x":    "registry.k8s.io/sig-storage/csi-resizer:v1.6.0",
+			">= 1.27.0": "registry.k8s.io/sig-storage/csi-resizer:v1.7.0",
+		},
+		OpenstackCSISnapshotter: {
+			"1.24.x":    "registry.k8s.io/sig-storage/csi-snapshotter:v5.0.1",
+			"1.25.x":    "registry.k8s.io/sig-storage/csi-snapshotter:v5.0.1",
+			"1.26.x":    "registry.k8s.io/sig-storage/csi-snapshotter:v6.1.0",
+			">= 1.27.0": "registry.k8s.io/sig-storage/csi-snapshotter:v6.2.1",
+		},
+		OpenstackCSISnapshotController: {
+			"1.24.x":    "registry.k8s.io/sig-storage/snapshot-controller:v5.0.1",
+			"1.25.x":    "registry.k8s.io/sig-storage/snapshot-controller:v5.0.1",
+			"1.26.x":    "registry.k8s.io/sig-storage/snapshot-controller:v6.1.0",
+			">= 1.27.0": "registry.k8s.io/sig-storage/snapshot-controller:v6.2.1",
```

Comment:
That's not going to work actually. Different CSI Snapshotter major/minor versions might have different CRDs. Difference in CRDs can introduce some very subtle bugs and issues. That's one of reasons why we use the same CSI Snapshotter version regardless of Kubernetes and CSI driver version.

Reply:
Didn't you said earlier that
> csi-snapshotter version must match the snapshot-controller version.

?!

Reply:
Yes, that's true, but what I meant is that we should use one version everywhere. Sorry for confusion!

Reply:
OK, changed that to a single version too